lang/solidity/Portfile

```
@@ -0,0 +1,86 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+
+PortGroup           github 1.0
+PortGroup           cmake 1.1
+PortGroup           boost 1.0
+
+github.setup        ethereum solidity 0.8.19 v
+github.tarball_from downloads
+distname            ${github.project}_${github.version}
+
+revision            0
+
+name                solidity
+categories          lang
+license             GPL-3+
+
+maintainers         {@kaimingguo kaiming.io:me} \
+                    openmaintainers
+
+description         Solidity, the Smart Contract Programming Language
+long_description    Solidity is a statically typed, contract-oriented, \
+                    high-level language for implementing smart contracts \
+                    on the Ethereum platform.
+homepage            https://soliditylang.org
+master_sites        https://github.com/ethereum/solidity/releases/download/v${github.version}/
+
+checksums           rmd160  df322fa402d7fc3d1c6643c0136a44251189ce28 \
+                    sha256  9bc195e1695f271b65326c73a12223377d52ae4b4fee10589c5e7bde6fa44194 \
+                    size    3239230
+
+set port_libfmt     libfmt9
+
+depends_build-append \
+                    port:${port_libfmt} \
+                    port:range-v3
+depends_lib         port:jsoncpp
+
+cmake.module_path-append \
+                    ${prefix}/lib/${port_libfmt}/cmake
+
+configure.args      -DTESTS:BOOL=OFF \
+                    -DBoost_USE_STATIC_LIBS:BOOL=OFF \
+                    -DWITH_Z3:BOOL=OFF \
+                    -DWITH_CVC4:BOOL=OFF \
+                    -DUSE_LD_GOLD:BOOL=OFF \
+                    -DRANGE_V3_INCLUDE_DIR:PATH="${prefix}/include" \
+                    -DJSONCPP_INCLUDE_DIR:PATH="${prefix}/include/jsoncpp" \
+                    -DJSONCPP_LIB_DIR:PATH="${prefix}/lib"
+
+patchfiles-append   patch-cmake_range-v3.cmake.diff \
+                    patch-libsolutil_JSON.cpp.diff
+
+default_variants    +z3
+
+variant z3 description {SMT Checker via Z3} {
+    depends_lib-append \
+        port:z3
+    configure.args-append \
+        -DSTRICT_Z3_VERSION:BOOL=OFF
+    configure.args-replace \
+        -DWITH_Z3:BOOL=OFF \
+        -DWITH_Z3:BOOL=ON
+}
+
+variant cvc4 description {SMT Checker via CVC4} {
+    depends_lib-append \
+        port:cvc4
+    configure.args-replace \
+        -DWITH_CVC4:BOOL=OFF \
+        -DWITH_CVC4:BOOL=ON
+}
+
+pre-test {
+    configure.args-replace \
+        -DTESTS:BOOL=OFF
+        -DTESTS:BOOL=ON
+}
```

You can't change configure arguments in pre-test; it's too late by that point. Phases are run in order: fetch, checksum, extract, patch, configure, build, test (if requested), destroot. By the time the `pre-test` block runs, the configure phase is long over.

This has no effect anyway since you haven't requested that the port allow tests to be run.

If tests can be run, but cannot run without configure having been run with `-DTESTS:BOOL=ON` and you don't want to use that unconditionally, then you'll have to relegate tests to a variant:

```suggestion
variant tests description {enable tests} {
    test.run        yes
    configure.args-replace \
        -DTESTS:BOOL=OFF
        -DTESTS:BOOL=ON
}
```